File: src/frontend/src/lib/types/tank.ts
import type { Position } from './position';
import { Direction } from './direction';
import { TankStatus } from './tankStatus';

export type Tank = {
	Name: string;
	Health: number;
	Position: Position;
	Status: TankStatus;
	PointRegister: number;
	DiedInTurn: number;
};

export function calculateRotationDegrees(direction: Direction, offset: number = 0): number {
	let rotation: number = offset;

	switch (direction) {
		default:
		case Direction.North:
			rotation += 270;
			break;
		case Direction.NorthEast:
			rotation += 315;
			break;
		case Direction.East:
			rotation += 0;
			break;
		case Direction.SouthEast:
			rotation += 45;
			break;
		case Direction.South:
			rotation += 90;
			break;
		case Direction.SouthWest:
			rotation += 135;
			break;
		case Direction.West:
			rotation += 180;
			break;
		case Direction.NorthWest:
			rotation += 225;
			break;
	}

	return rotation;
}

export function calculateRotationRadians(direction: Direction, offsetDegree: number = 0): number {
	return (calculateRotationDegrees(direction, offsetDegree) * Math.PI) / 180;
}

File: src/backend/BotBattle.Core/Tank.cs
using BotBattle.Core.Enums;

namespace BotBattle.Core;

public class Tank
{
    public required string Name { get; init; }
    public Position Position { get; set; } = new(0, 0);
    public int Health { get; set; } = 1000;
    public TankStatus Status { get; set; } = TankStatus.Alive;
    public int PointRegister { get; set; } = 0;
    public int DiedInTurn { get; set; } = -1;
    public WeaponSystem WeaponSystem { get; set; }
}